clients/java/signalr/src/main/java/com/microsoft/aspnet/signalr/NegotiateResponse.java

```diff
@@ -15,10 +15,15 @@
     private Set<String> availableTransports = new HashSet<>();
     private String redirectUrl;
     private String accessToken;
+    private String error;
     private JsonParser jsonParser = new JsonParser();
 
     public NegotiateResponse(String negotiatePayload) {
         JsonObject negotiateResponse = jsonParser.parse(negotiatePayload).getAsJsonObject();
+        if (negotiateResponse.has("error")) {
+            this.error = negotiateResponse.get("url").getAsString();
+            return;
```

Comment:
I don't think the parsing logic should exit early like this. You're right that an `error` property generally means everything else should be ignored, but let the caller decide that.